ddtrace/contrib/asgi/middleware.py

```diff
@@ -58,6 +60,13 @@ def _extract_tags_from_scope(scope):
     return tags
 
 
+def _extract_headers(scope):
+    headers = scope.get("headers")
+    if headers:
+        # headers: (Iterable[[byte string, byte string]])
+        return dict((bytes_to_str(k), bytes_to_str(v)) for (k, v) in headers)
+
```

Comment:
Last quick item - now it’s _theoretically_ possible (also not covered by the ASGI spec) that `headers` be `None`, is that supported? We might want to just default to `{}` instead. 

```suggestion
def _extract_headers(scope):
    headers = scope.get("headers")
    if headers:
        # headers: (Iterable[[byte string, byte string]])
        return dict((bytes_to_str(k), bytes_to_str(v)) for (k, v) in headers)
    return {}
```